Mage.Client/src/main/java/mage/client/game/GamePanel.java

```diff
@@ -962,6 +939,19 @@ public synchronized void updateGame() {
             exileWindow.loadCardsAndShow(exile, bigCard, gameId);
         }
 
+        // Remove empty ExileView lists, which prevent removing old windowDialogs after rollbacks.
+        exileViewList.removeIf(HashMap::isEmpty);
+
+        // Remove old exiles entries and windowDialogs, which can prevent generating new windowDialogs after rollbacks.
+        if (exileViewList.isEmpty()) {
+            Iterator<CardInfoWindowDialog> iterator = exiles.values().iterator();
```

Comment:
I think it can be improved for all use cases with multiple zones too (if you have one empty and one normal windows) — you can iterate current exile windows, search it in game view and clean on miss id.